### ISSUE TITLE
signal flow graph

### ISSUE THREAD

**Issue by tomara-x** [tomara-x/witches#23]
*the sound of my exploding brain*

**Comment by tomara-x** [comment]
I wanna stitch a taphy to inlets and outlets and try self patching in daemon mode

**Comment by tomara-x** [comment]
oh past Amy! you didn't know what you were doing!

**Comment by tomara-x** [comment]
now I'm just doing this:
```
instr patch
ktrg metro 1/4
if ktrg == 1 then
    turnoff2(99, 0, 0)
    schedulek(99, 0, -1)
endif
endin
```
and just globalize any variables I wanna use as in/outlets, connect those in instr 99, and redefine instr 99 in live mode when I need to change these connections.

**Comment by tomara-x** [comment]
going for a "gr_instr_var" naming convention for those global vars. r being the signal rate

**Comment by tomara-x** [comment]
I might just make a mixer instrument like this! much easier than trying to figure out the ops!

**Comment by tomara-x** [comment]
> I could use a global array for different channels if I need more than one instance of the same instrument running at the same time (specifying which channel...

this is to avoid the way a globa variable gets overwritten by newer instances (test files)

**Comment by tomara-x** [comment]
this one's good for now, I think. I wanna get deeper into the instr and opcode stuff, also I think I'm dropping the score and sticking with the real-time orc events. something like a schedule instrument, you get all the loopy, conditionally, arrayy goodness of the orc and use it as your timeline! I still like the score though, it's pretty! it's like a tracker and I love those!

I learned so much in this one, the scheduling, the instr name-num conversion, the live coding biz, how powerful the instrument is, and how much is possible when stitching many of them together (I started seeing that opcodes resemble the opcodes of a processore, and the instruments are like functions of a programming language. guess they shoulda called it assemblysound!) and boy am I glad I implemented the girls as opcodes! kinda like templates and I can write instruments that use them in any way that fits the situation. (on paper the very first plan was gonna make taphy an instr.. so glad I didn't do that) I'm picking up some granular too after I'm done getting used to the rt-scheduling biz.

but yeah, live coding and modular system stuff figured out, and now.. don't know, feeling a block lately.. maybe I should take a break
